### PR TITLE
Switch to nvm version when changing directory to a directory with .nvmrc

### LINKED ISSUE #19: Use .nvmrc automatically when changing directory.

### PR DESCRIPTION
**Pull request by DropsOfSerenity**
Closes #19

### CODE REVIEW

File: zshrc [hunk 1]
```diff
@@ -78,6 +78,30 @@ export PATH="/usr/local/bin:$PATH"
...
+  local nvmrc_path;
+  node_version="$(nvm version)"
+  nvmrc_path="$(nvm_find_nvmrc)"
+
+  if [ -n "$nvmrc_path" ]; then
+    local nvmrc_node_version;
+    nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
+
+    if [ "$nvmrc_node_version" = "N/A" ]; then
+      nvm install
```
Comment (DropsOfSerenity) [inline zshrc:93]: I think there is a discussion to be had here, we may rather just inform the user, as I could see this behavior being rather annoying.

Comment (michaelachrisco) [reply zshrc:93]: I can see this causing some confusion from developers that are not expecting autoload node versioning. 

Just spitballing, what do you think of adding an env var to disable? If it becomes annoying, they have the option of turning it off (like like `npm sponsor`).

**Review by michaelachrisco — APPROVED** [review]
Code looks good (nice use of autoload). Added comments regarding the autoloading.

**Review by TomAlperin — APPROVED** [review]
Looks good but I would definitely include options to control what is installed.

File: zshrc (hunk 1 above)
Comment (DropsOfSerenity) [reply zshrc:97]: They can customize their option fully in `~/.zshrc.user` for example if they do not want this behavior they can disable the hook entirely: `add-zsh-hook -d chpwd load-nvmrc`

**Comment by DropsOfSerenity** [comment]
I've reconsidered this and think this behavior is too intrusive. I think there is no need to wrap nvm as it does a great job managing node versions on it's own. For projects that don't use docker, nvm is a tool people will need to learn anyway, we can do a better job at incorporating steps into the `README` of the various projects. For projects the *do* use docker, nvm is not really needed as docker is your node version manager.

**Pull request closed without merging**